### ISSUE TITLE
HW: Volume deletion fails but e2e reports SUCCESS

### ISSUE THREAD

**Issue by okartau** [intel/pmem-csi#337]
This happens when testing code of #336 on HW-based node, using `make test_e2e`
There will be a problem described in #317: after umount returns in pmem-csi node, file system remains mounted in host. Next shred corrupts the contents and host will look having zetabyte volume mounted. Following lvremove fails as in below trace:

```
pmem-csi-node-vh7pz/pmem-driver: I0705 20:19:14.947353 NodeUnStageVolume: VolumeID:2860cf11-9f62-11e9-a266-962c6531b78c VolumeName: Staging target path:/var/lib/kubelet/plugins/kubernetes.io/csi/pv/pmem-sanity-staging.AokfPi
pmem-csi-node-vh7pz/pmem-driver: I0705 20:19:14.950777 NodeUnstageVolume: detected mountedDev: /dev/ndbus0region0fsdax/2860cf11-9f62-11e9-a266-962c6531b78c
pmem-csi-controller-0/pmem-driver: I0705 20:19:14.954037 GRPC call: /csi.v1.Controller/DeleteVolume
pmem-csi-node-vh7pz/pmem-driver: I0705 20:19:14.950786 NodeUnStageVolume: umount /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pmem-sanity-staging.AokfPi
pmem-csi-node-vh7pz/pmem-driver: I0705 20:19:14.953104 GRPC response: {}
pmem-csi-controller-0/pmem-driver: I0705 20:19:14.954050 GRPC request: {"volume_id":"2860cf11-9f62-11e9-a266-962c6531b78c"}
pmem-csi-controller-0/pmem-driver: I0705 20:19:14.954435 DeleteVolume: volumeID: 2860cf11-9f62-11e9-a266-962c6531b78c
pmem-csi-controller-0/pmem-driver: I0705 20:19:14.954442 Connecting to node controller : tcp://10.237.71.95:10001
pmem-csi-controller-0/pmem-driver: I0705 20:19:14.954449 Connecting to 10.237.71.95:10001
pmem-csi-node-vh7pz/pmem-driver: I0705 20:19:14.958235 GRPC call: /csi.v1.Controller/DeleteVolume
pmem-csi-node-vh7pz/pmem-driver: I0705 20:19:14.958245 GRPC request: {"volume_id":"2860cf11-9f62-11e9-a266-962c6531b78c"}
pmem-csi-node-vh7pz/pmem-driver: I0705 20:19:14.958658 DeleteVolume: volumeID: 2860cf11-9f62-11e9-a266-962c6531b78c
pmem-csi-node-vh7pz/pmem-driver: I0705 20:19:14.958668 ClearDevice: path: /dev/ndbus0region0fsdax/2860cf11-9f62-11e9-a266-962c6531b78c flush:true
pmem-csi-node-vh7pz/pmem-driver: I0705 20:19:14.958682 Wiping entire device: /dev/ndbus0region0fsdax/2860cf11-9f62-11e9-a266-962c6531b78c
pmem-csi-node-vh7pz/pmem-driver: I0705 20:19:14.958690 Executing: shred -n 1 /dev/ndbus0region0fsdax/2860cf11-9f62-11e9-a266-962c6531b78c
pmem-csi-node-vh7pz/pmem-driver: I0705 20:19:15.011265 Executing: lvremove -fy /dev/ndbus0region0fsdax/2860cf11-9f62-11e9-a266-962c6531b78c
cleanup: error: DeleteVolume: rpc error: code = Internal desc = Failed to delete volume 2860cf11-9f62-11e9-a266-962c6531b78c rpc error: code = Internal desc = Failed to delete volume: exit status 5
[AfterEach] pmem csi
  /home/okartau/go/src/github.com/intel/pmem-csi/vendor/github.com/kubernetes-csi/csi-test/pkg/sanity/tests.go:51
pmem-csi-node-vh7pz/pmem-driver: I0705 20:19:15.060655 Output:   Logical volume ndbus0region0fsdax/2860cf11-9f62-11e9-a266-962c6531b78c contains a filesystem in use.
pmem-csi-controller-0/pmem-driver: E0705 20:19:15.061034 GRPC error: rpc error: code = Internal desc = Failed to delete volume 2860cf11-9f62-11e9-a266-962c6531b78c rpc error: code = Internal desc = Failed to delete volume: exit status 5
Jul  5 23:19:15.064: INFO: ExecWithOptions {Command:[rmdir /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pmem-sanity-target.jkciEk] Namespace:default PodName:pmem-csi-node-testing-j26zp ContainerName :socat Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
Jul  5 23:19:15.265: INFO: Unexpected error occurred: command terminated with exit code 1
```

This is all as in #317 and needs fixing.
But this issue I create to resolve this: Why at the end of that e2e we see:
```
Ran 41 of 90 Specs in 180.486 seconds
SUCCESS! -- 41 Passed | 0 Failed | 1 Pending | 48 Skipped
--- PASS: TestE2E (180.49s)
PASS
ok      github.com/intel/pmem-csi/test/e2e      180.564s
```

Recently there was fix to propagate this failure from node to controller.
Does this work?
Why the failure is not propagating as e2e failure ?

**Comment by avalluri** [comment]
> Why the failure is not propagating as e2e failure ?
Because those failures were occured while deleting volumes from cleanup, which just logs the errors,

**Comment by pohly** [comment]
Amarnath Valluri <notifications@github.com> writes:

>> Why the failure is not propagating as e2e failure ?
> Because those failures were occured while deleting volumes from
> cleanup, which just logs the errors,

If that DeleteVolume came from the cleanup code (are you sure it was?),
then the test must have failed already, because a successful test
shouldn't leave any volumes behind.

Is the full log available?

**Comment by avalluri** [comment]
> Amarnath Valluri <notifications@github.com> writes:
> > Why the failure is not propagating as e2e failure ?
>  Because those failures were occured while deleting volumes from
>  cleanup, which just logs the errors,
> If that DeleteVolume came from the cleanup code (are you sure it was?),

```
cleanup: error: DeleteVolume: rpc error: code = Internal desc = Failed to delete volume 2860cf11-9f62-11e9-a266-962c6531b78c rpc error: code = Internal desc = Failed to delete volume: exit status 5
```
This log line shows that failure was coming from cleanup code. I my self didn't see full logs.

**Comment by pohly** [comment]
> If that DeleteVolume came from the cleanup code (are you sure it was?), then the test must have failed already, because a successful test shouldn't leave any volumes behind.

That's true for the stress test. However, here a reboot test got skipped ("node "xxxx" not one of the expected QEMU nodes (worker<number>))"), and then cleaning up after that test failed.

**Comment by avalluri** [comment]
The root cause of the problem was that failure of DeleteVolume was invoked from cleanup code, which got fixed in #336, So we can close this issue.
